### DISCUSSION TITLE
go-ceph build error

### DISCUSSION THREAD

**Discussion by QZB349549482** [ceph/go-ceph#835] [Q&A] [answered]
My ceph version is quincy and go-ceph verison is v0.19.0. It means that  I need build go binary like this 
```
go build -tags quincy,ceph_preview xxxx.go
```
But I have got a vendor error 
```
# github.com/ceph/go-ceph/rbd
../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:35:50: could not determine kind of name for C.RBD_ENCRYPTION_ALGORITHM_AES128
../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:36:50: could not determine kind of name for C.RBD_ENCRYPTION_ALGORITHM_AES256
../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:73:19: could not determine kind of name for C.RBD_ENCRYPTION_FORMAT_LUKS1
../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:87:19: could not determine kind of name for C.RBD_ENCRYPTION_FORMAT_LUKS2
../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:29:26: could not determine kind of name for C.rbd_encryption_algorithm_t
../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:110:9: could not determine kind of name for C.rbd_encryption_format
../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:22:11: could not determine kind of name for C.rbd_encryption_format_t
../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:134:9: could not determine kind of name for C.rbd_encryption_load
../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:64:12: could not determine kind of name for C.rbd_encryption_luks1_format_options_t
../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:78:12: could not determine kind of name for C.rbd_encryption_luks2_format_options_t
../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:23:11: could not determine kind of name for C.rbd_encryption_options_t
../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:71:30: could not determine kind of name for C.sizeof_rbd_encryption_luks1_format_options_t
../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:85:30: could not determine kind of name for C.sizeof_rbd_encryption_luks2_format_options_t
```
Is there something wrong I have done .

**Comment by anoopcs9** [comment]
> But I have got a vendor error
> 
> ```
> # github.com/ceph/go-ceph/rbd
> ../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:35:50: could not determine kind of name for C.RBD_ENCRYPTION_ALGORITHM_AES128
> ../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:36:50: could not determine kind of name for C.RBD_ENCRYPTION_ALGORITHM_AES256
> ../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:73:19: could not determine kind of name for C.RBD_ENCRYPTION_FORMAT_LUKS1
> ../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:87:19: could not determine kind of name for C.RBD_ENCRYPTION_FORMAT_LUKS2
> ../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:29:26: could not determine kind of name for C.rbd_encryption_algorithm_t
> ../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:110:9: could not determine kind of name for C.rbd_encryption_format
> ../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:22:11: could not determine kind of name for C.rbd_encryption_format_t
> ../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:134:9: could not determine kind of name for C.rbd_encryption_load
> ../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:64:12: could not determine kind of name for C.rbd_encryption_luks1_format_options_t
> ../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:78:12: could not determine kind of name for C.rbd_encryption_luks2_format_options_t
> ../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:23:11: could not determine kind of name for C.rbd_encryption_options_t
> ../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:71:30: could not determine kind of name for C.sizeof_rbd_encryption_luks1_format_options_t
> ../../vendor/github.com/ceph/go-ceph/rbd/encryption.go:85:30: could not determine kind of name for C.sizeof_rbd_encryption_luks2_format_options_t
> ```
> 
> Is there something wrong I have done .

Can you please check whether corresponding RBD development packages are installed? Depending upon the distro it might be named **_librbd-devel_** or **_librbd-dev_**. For example in RPM based distros following command can help you in checking the RBD development packages installed:

```
# rpm -q librbd-devel
```

**Reply by QZB349549482** [reply]
It is  work that I change the tag from pacific to octopus  or earlier version like nautilus . The librbd-dev version is 15.2.17-0ubuntu0.20.04.1 . Thank you !

**Reply by anoopcs9** [reply, marked as answer]
_librdb-dev_ is of version 15.x which is equivalent to **octopus** and NOT quincy. May be those encryption structures and APIs are not yet available with **octopus** version of librbd.

**Reply by QZB349549482** [reply]
OK. I got it . Thank you !

**Reply by phlogistonjohn** [reply]
I often use https://docs.ceph.com/en/latest/releases/index.html to map between the version numbers and the code names. This page is usually up to date.

**Comment by anoopcs9** [comment]
> My ceph version is quincy and go-ceph verison is v0.19.0. It means that I need build go binary like this
> 
> ```
> go build -tags quincy,ceph_preview xxxx.go

What are you trying to achieve here? Can you please paste the exact command used?
